specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/stable/2022-07-01/extensions.json

```diff
@@ -0,0 +1,691 @@
+        },
+        "time": {
+          "type": "string",
+          "description": "DateLiteral (per ISO8601) noting the time of installation status."
+        }
+      }
+    },
+    "Extension": {
+      "description": "The Extension object.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "Properties of an Extension resource",
+          "properties": {
+            "extensionType": {
+              "description": "Type of the Extension, of which this resource is an instance of.  It must be one of the Extension Types registered with Microsoft.KubernetesConfiguration by the Extension publisher.",
+              "type": "string"
+            },
+            "autoUpgradeMinorVersion": {
+              "description": "Flag to note if this extension participates in auto upgrade of minor version, or not.",
+              "type": "boolean",
+              "default": true
+            },
+            "releaseTrain": {
+              "description": "ReleaseTrain this extension participates in for auto-upgrade (e.g. Stable, Preview, etc.) - only if autoUpgradeMinorVersion is 'true'.",
+              "type": "string",
+              "default": "Stable"
+            },
+            "version": {
+              "description": "User-specified version of the extension for this extension to 'pin'. To use 'version', autoUpgradeMinorVersion must be 'false'.",
+              "type": "string",
+              "x-nullable": true
+            },
+            "scope": {
+              "description": "Scope at which the extension is installed.",
+              "$ref": "#/definitions/Scope"
+            },
+            "configurationSettings": {
+              "description": "Configuration settings, as name-value pairs for configuring this extension.",
+              "type": "object",
+              "x-nullable": true,
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            "configurationProtectedSettings": {
+              "description": "Configuration settings that are sensitive, as name-value pairs for configuring this extension.",
+              "type": "object",
+              "x-nullable": true,
+              "x-ms-secret": true,
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            "installedVersion": {
+              "description": "Installed version of the extension.",
+              "type": "string",
+              "readOnly": true,
+              "x-nullable": true
+            },
+            "provisioningState": {
+              "description": "Status of installation of this extension.",
+              "type": "string",
+              "readOnly": true,
+              "$ref": "../../common/2022-07-01/definitions.json#/definitions/ProvisioningStateDefinition"
```

Comment:
>07 [](http://example.com/codeflow?start=41&length=2)

Like I said in my other comment most teams do not rev the common definitions in each iterations to avoid changes like this. Not a blocking comment but just something to consider for future iterations

Reply:
Addressed in another comment

Reply:
@raosuhas How do teams normally versions these directories then? Versioning with api-version seems a little weird since we now have 2022-03-01 common directory tracking for 2022-07-01 version? Are teams doing something like `v1`, `v2`, etc.?